clarifai/runners/dockerfile_template/Dockerfile.nim

```diff
@@ -0,0 +1,71 @@
+FROM nvcr.io/nim/meta/llama-3.1-8b-instruct:1.1.2 as build
+
+FROM gcr.io/distroless/python3-debian12:debug
+
+
+COPY --from=build /bin/bash /bin/bash
+COPY --from=build /bin/sh /bin/sh
+
+# we have to overwrite the python3 binary that the distroless image uses
+COPY --from=build /opt/nim/llm/.venv/bin/python3.10 /usr/bin/python3
+COPY --from=build /opt/nim/llm/.venv/bin/python3.10 /usr/local/bin/python3.10
+
+# also copy in all the lib files for it.
+COPY --from=build /lib /lib
+COPY --from=build /lib64 /lib64
+COPY --from=build /usr/lib/ /usr/lib/
+COPY --from=build /usr/local/lib/ /usr/local/lib/
+# ldconfig is needed to update the shared library cache so system libraries (like CUDA) can be found
+COPY --from=build /usr/sbin/ldconfig /sbin/ldconfig
+COPY --from=build /usr/sbin/ldconfig.real /sbin/ldconfig.real
+COPY --from=build /etc/ld.so.conf /etc/ld.so.conf
+COPY --from=build /etc/ld.so.cache /etc/ld.so.cache
+COPY --from=build /etc/ld.so.conf.d/ /etc/ld.so.conf.d/
+
+# COPY NIM files
+COPY  --from=build /opt /opt
+COPY  --from=build /etc/nim /etc/nim
+
+# Set environment variables to use the nim libraries and python
+ENV PYTHONPATH=${PYTHONPATH}:/opt/nim/llm/.venv/lib/python3.10/site-packages:/opt/nim/llm
+ENV PATH="/opt/nim/llm/.venv/bin:/opt/hpcx/ucc/bin:/opt/hpcx/ucx/bin:/opt/hpcx/ompi/bin:$PATH"
+
+ENV LD_LIBRARY_PATH="/opt/hpcx/ucc/lib/ucc:/opt/hpcx/ucc/lib:/opt/hpcx/ucx/lib/ucx:/opt/hpcx/ucx/lib:/opt/hpcx/ompi/lib:/opt/hpcx/ompi/lib/openmpi:/opt/nim/llm/.venv/lib/python3.10/site-packages/tensorrt_llm/libs:/opt/nim/llm/.venv/lib/python3.10/site-packages/nvidia/cublas/lib:/opt/nim/llm/.venv/lib/python3.10/site-packages/tensorrt_libs:/opt/nim/llm/.venv/lib/python3.10/site-packages/nvidia/nccl/lib:$LD_LIBRARY_PATH"
```

Comment:
are all of these needed?

Reply:
Not sure about all of these, but I think will need to include which are at the path  `/opt/`

Reply:
btw I look into all the environment variables from the NIM image and then included into this image